### PR TITLE
 Fix in AutoML: Wait Only retrieve associations fields once

### PR DESCRIPTION
**Pull request by unmonoqueteclea**
I already wrote this in July, but I don't remember I never merged it. It solves a problem reported recently by a user.

This solves two problems:

- sometimes we don't wait associations to finish before retrieving their fields
- non-prefered fields are not shown in filtered datasets

### CODE REVIEW

File: automl/automl-library/library.whizzml [hunk 1]
```diff
@@ -86,7 +86,7 @@
 
 ;; Set a list of fields from a dataset as non-preferred
 (define (set-non-preferred dataset-id field-names)
-  (let (fids (field-ids-from-names field-names dataset-id)
+  (let (fids (field-ids-from-names field-names (wait dataset-id))
```
Comment (mmerce) [inline automl/automl-library/library.whizzml]: wouldn't it be better to wait inside the field-ids-from-names function?

File: automl/automl-script/script.whizzml [hunk 2]
```diff
@@ -68,7 +68,10 @@
 (define non-preferred-fields
   (remove-duplicates
    (flatten
-    (map (lambda(ds) (field-names-from-ids (non-preferred ds) ds))
+    (map (lambda(ds)
+           (if (and ds (> (count (resource-id ds)) 0))
+             (field-names-from-ids (non-preferred (wait ds)) ds)
```
Comment (mmerce) [inline automl/automl-script/script.whizzml]: Same as before, maybe waiting in non-preferred procedure...

File: automl/automl-script/script.whizzml [hunk 3]
```diff
@@ -212,9 +216,10 @@
          (when (= "dataset" (resource-type ds))
            (let (name (str (resource-name ds) " | filtered")
                  fields (field-ids-from-names selected-fields ds))
-             (create-dataset {"name" name
-                              "origin_dataset" ds
-                              "input_fields" (remove-duplicates fields)}))))
+             (set-non-preferred
+              (create-dataset {"name" name  "origin_dataset" ds
+                              "input_fields" (remove-duplicates fields)})
```
Comment (mmerce) [inline automl/automl-script/script.whizzml]: indentation

**Comment by mmerce** [comment]
Just a couple of nits.

**Comment by unmonoqueteclea** [comment]
Done!

**Review by mmerce — APPROVED** [review]

**Pull request merged**